src/Illuminate/Console/Command.php

```diff
@@ -189,7 +204,70 @@ public function run(InputInterface $input, OutputInterface $output)
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        return $this->laravel->call([$this, 'handle']);
+        if ($this->hasValidInput()) {
+            return $this->laravel->call([$this, 'handle']);
+        }
+
+        $this->displayValidationErrors();
+    }
+
+    /**
+     * Check the argument and option values are valid.
+     *
+     * @return bool
+     *
+     * @throws \Illuminate\Contracts\Container\BindingResolutionException
+     */
+    protected function hasValidInput()
+    {
+        if (! isset($this->rules)) {
+            return true;
+        }
+
+        $this->validator = $this->validator ?? $this->laravel->make(Validator::class)->make(
+            array_merge($this->arguments(), $this->options()),
+            $this->rules,
+            $this->messages(),
+            $this->attributes()
+        );
+
+        return ! $this->validator->fails();
+    }
+
+    /**
+     * Get the custom error messages for console command validation.
+     *
+     * @return array
+     */
+    protected function messages()
+    {
+        return [];
+    }
+
+    /**
+     * Get the custom attribute names for console command validation.
+     *
+     * @return array
+     */
+    protected function attributes()
+    {
+        return [];
+    }
+
+    /**
+     * Display the validation error messages for invalid arguments and options.
+     *
+     * @return void
+     */
+    protected function displayValidationErrors()
+    {
+        $this->info('Argument / option validation failed. Please check the error messages below:');
+
+        foreach ($this->validator->errors()->all() as $error) {
+            $this->error($error);
+        }
+
+        exit(1);
```

Comment:
We can't call exit here.

Reply:
return 1?

Reply:
Or forget the return altogether and let it be implicit?

Reply:
Usually commands `return 1` in case of error. And then you need to change return type from `void` to `int` I suppose.

Reply:
But it looks like wrong to return integer from `displayValidationErrors` method.

Reply:
One could even consider simply throwing an exception.